File: RedditPictureReader/app/src/main/java/se/slotte/ludwig/redditpicturereader/picture_feed/PictureView.java
package se.slotte.ludwig.redditpicturereader.picture_feed;

import java.util.List;

import se.slotte.ludwig.redditpicturereader.picture_feed.data.model.Children;

/**
 * Created by ludwigslotte on 16/10/16.
 */

public interface PictureView {
    void onNetworkError();

    void onSuccess(List<Children> success);

    void loadPictures();
}

File: RedditPictureReader/app/src/main/java/se/slotte/ludwig/redditpicturereader/picture_feed/PictureView.kt
package se.slotte.ludwig.redditpicturereader.picture_feed

import se.slotte.ludwig.redditpicturereader.picture_feed.data.model.Children

/**
 * Created by ludwigslotte on 16/10/16.
 */

interface PictureView {
    fun onNetworkError()

    fun onSuccess(success: List<Children>)

    fun startLoading()

    fun onGeneralError()

    fun setOnErrorClickListener()
}
